### ISSUE TITLE
Working within clusters

### ISSUE THREAD

**Issue by hegza** [stm32-rs/svdtools#131]
TL;DR: most svdtools features don't work within the `<cluster>` tag. Should they?

### Background

Somewhat similar to https://github.com/stm32-rs/svdtools/pull/27, our platform SVDs have registers organized into clusters like so:

```xml
<peripheral>
  <name>SUBSYSTEM</name>
  <cluster>
    <name>PERIPHERAL_1</name>
    <description />
    <register>REG_1</register>
    <register>REG_2.</register>
  </cluster>
  <cluster>
    <name>PERIPHERAL_2name>
    <description />
    <register>...</register>
    <register>...</register>
  </cluster>
</peripheral>
```

### Problem

I have no idea whether it's a good idea to arrange the device like so but in case one does, we encounter issues with `svd patch` which seems to be unable to make **most** types of changes inside clusters. 

Take for instance `_array`. If you were to write the following patch.yaml against the above architecture:

```yaml
"SUBSYSTEM":
  _array:
    REG_*: {}
```

you would get the following error:

> File "src/lxml/etree.pyx", line 943, in lxml.etree._Element.remove
> ValueError: Element is not a child of this node.

while if you didn't have the cluster layer, the operation would pass successfully. I suspect the `svd patch` code is looking for the matching "REG_*" symbols as leaf of the `<peripheral>SUBSYSTEM</peripheral>`-layer, while they cannot be found due to the symbols actually being wrapped one layer deeper inside a `<cluster>`-tag. However, we cannot currently represent the cluster-layer in the YAML syntax. https://github.com/stm32-rs/svdtools/pull/27 seems to work around this through clever re-use of `_cluster` tag.

### Follow up

Did I get the problem statement right? How do you feel about the issue? Do you think it's a user problem i.e. too weird SVD layout, or would it be appropriate for `svd patch` to support changes inside clusters.

If you think this is in the _"should fix"_ category, I could investigate if I could come up with a PR for it. In that case I'd need at least an opinion on if the cluster-layer could be represented in the YAML patch syntax similar to the peripheral, register and field layers, like so:

```yaml
"PERIPHERAL":
  CLUSTER:
    REGISTER:
      _modify:
        FIELD:
          description: "a description"
```

**Comment by burrbull** [comment]
Closing as completed. Reopen if still needs improve.
